lib/widgets/barcode_scanner.dart

```diff
@@ -7,8 +7,8 @@ import 'package:mlkit_scanner/platform/ml_kit_channel.dart';
 import 'package:mlkit_scanner/widgets/camera_preview.dart';
 
 /// Signature of the BarcodeScanner success initialize scanner function.
-typedef BarcodeScannerInitializeCallback = void Function(
-    BarcodeScannerController? controller);
+typedef BarcodeScannerInitializeCallback = Function(
```

Comment:
According to such a signature, you can pass an asynchronous callback, void type is automatically packed by dart into a Future.

You can check this example in dartpad:
```
import 'package:flutter/material.dart';

class TestWidget extends StatelessWidget {
  
  final void Function(bool flag) nonAsyncCallback;
  
  const TestWidget(this.nonAsyncCallback);
  
  @override
  Widget build(BuildContext context) {
    nonAsyncCallback(true);
    return const Text("Hello");
  }
}

void main() {
  runApp(
    MaterialApp(
      home: TestWidget(
        (flag) async {
          await Future.delayed(const Duration(seconds: 5));
          print(flag);
        },
      ),
    ),
  );
}
```


Btw, I agree about the controller changes, it cannot be null in the context.